### PR TITLE
Integrate the new prebuild image notebooks into upstream

### PR DESCRIPTION
**Pull request by atheo89**
This PR includes all the necessary changes to switch from jupyterhub to jupyter notebooks. The corresponding open issue is the https://github.com/opendatahub-io/odh-manifests/issues/671

## Description

The old Thoth images from `jupyterhub/notebook-images/overlays/additional` are replaced with the new set of notebooks into `notebook-images/base`.

Below is the folder tree of the new folder structure under the root path of the repository. 
```
├── notebook-images
│   └── base
│       ├── jupyter-datascience-notebook-imagestream.yaml
│       ├── kustomization.yaml
│       ├── jupyter-minimal-gpu-notebook-imagestream.yaml
│       ├── jupyter-minimal-notebook-imagestream.yaml
│       ├── jupyter-pytorch-notebook-imagestream.yaml
│       └── jupyter-tensorflow-notebook-imagestream.yaml
```
These images are stored in (https://github.com/opendatahub-io/notebooks) and are built with Openshift CI and mirrored to Quay (https://quay.io/repository/opendatahub/notebooks).

Updated testing KfDef `tests/setup/odh-core.yaml` to use the `notebook-images` folder

Last, as part of this transition, refactored the test notebooks `tests/resources/notebook-controller/notebooks` as presented below:
```
├── tests
│  ...
│   ├── resources
│   │   ...
│   │   ├── notebook-controller
│   │   │   ├── notebooks
│   │   │   │   ├── cuda-jupyter-minimal-ubi8-python-3-8.yaml
│   │   │   │   ├── cuda-jupyter-pytorch-ubi8-python-3-8.yaml
│   │   │   │   ├── cuda-jupyter-tensorflow-ubi8-python-3-8.yaml
│   │   │   │   ├── jupyter-datascience-ubi8-python-3-8.yaml
│   │   │   │   ├── jupyter-minimal-ubi8-python-3-8.yaml
```

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: jupyter-notebooks/base/generic-data-science-notebook-imagestream.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+apiVersion: image.openshift.io/v1
+kind: ImageStream
+metadata:
+  labels:
+    opendatahub.io/notebook-image: "true"
+  annotations:
+    opendatahub.io/notebook-image-url: "https://github.com/red-hat-data-services/notebooks/tree/main/jupyter/datascience/ubi8-python-3.8"
```
Comment (samuelvl) [inline jupyter-notebooks/base/generic-data-science-notebook-imagestream.yaml:7]:
```suggestion
    opendatahub.io/notebook-image-url: "https://github.com/opendatahub-io/notebooks/tree/main/jupyter/datascience/ubi8-python-3.8"
```

File: jupyter-notebooks/base/generic-data-science-notebook-imagestream.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    opendatahub.io/notebook-image-name: "Standard Data Science"
+    opendatahub.io/notebook-image-desc: "Jupyter notebook image with a set of data science libraries that advanced AI/ML notebooks will use as a base image to provide a standard for libraries avialable in all notebooks"
+    opendatahub.io/notebook-image-order: "20"
+  name: s2i-generic-data-science-notebook
```
Comment (samuelvl) [inline jupyter-notebooks/base/generic-data-science-notebook-imagestream.yaml:11]:
```suggestion
  name: jupyter-datascience-notebook
```

File: jupyter-notebooks/base/generic-data-science-notebook-imagestream.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+apiVersion: image.openshift.io/v1
+kind: ImageStream
+metadata:
+  labels:
+    opendatahub.io/notebook-image: "true"
+  annotations:
+    opendatahub.io/notebook-image-url: "https://github.com/red-hat-data-services/notebooks/tree/main/jupyter/datascience/ubi8-python-3.8"
+    opendatahub.io/notebook-image-name: "Standard Data Science"
```
Comment (samuelvl) [inline jupyter-notebooks/base/generic-data-science-notebook-imagestream.yaml:8]:
```suggestion
    opendatahub.io/notebook-image-name: "Jupyter Data Science"
```

File: jupyter-notebooks/base/generic-data-science-notebook-imagestream.yaml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  lookupPolicy:
+    local: true
+  tags:
+  - annotations:
+      opendatahub.io/notebook-software: '[{"name":"Python","version":"v3.8"}]'
+      opendatahub.io/notebook-python-dependencies: '[{"name":"Boto3","version":"1.17"},{"name":"Kafka-Python","version":"2.0"},{"name":"Matplotlib","version":"3.4"},{"name":"Numpy","version":"1.19"},{"name":"Pandas","version":"1.2"},{"name":"Scikit-learn","version":"0.24"},{"name":"Scipy","version":"1.6"}]'
+      openshift.io/imported-from: quay.io/opendatahub/notebooks
+    from:
+      kind: DockerImage
+      # if you change the image tag, change the "-N" on the tag name immediately below
```
Comment (samuelvl) [inline jupyter-notebooks/base/generic-data-science-notebook-imagestream.yaml:22]: This comment doesn't apply anymore:

```suggestion
```

File: jupyter-notebooks/base/minimal-gpu-notebook-imagestream.yaml [hunk 5]
```diff
@@ -0,0 +1,27 @@
+apiVersion: image.openshift.io/v1
+kind: ImageStream
+metadata:
+  labels:
+    opendatahub.io/notebook-image: "true"
+  annotations:
+    opendatahub.io/notebook-image-url: "https://github.com/red-hat-data-services/notebooks/tree/main/jupyter/minimal/ubi8-python-3.8"
```
Comment (samuelvl) [inline jupyter-notebooks/base/minimal-gpu-notebook-imagestream.yaml:7]:
```suggestion
    opendatahub.io/notebook-image-url: "https://github.com/opendatahub-io/notebooks/tree/main/jupyter/minimal/ubi8-python-3.8"
```

File: jupyter-notebooks/base/minimal-notebook-imagestream.yaml [hunk 6]
```diff
@@ -0,0 +1,28 @@
+apiVersion: image.openshift.io/v1
```
Comment (samuelvl) [inline jupyter-notebooks/base/minimal-notebook-imagestream.yaml:1]:
```suggestion
---
apiVersion: image.openshift.io/v1
```

File: jupyter-notebooks/base/minimal-notebook-imagestream.yaml [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+kind: ImageStream
+metadata:
+  labels:
+    opendatahub.io/notebook-image: "true"
+  annotations:
+    opendatahub.io/notebook-image-url: "https://github.com/red-hat-data-services/notebooks/tree/main/jupyter/minimal/ubi8-python-3.8"
+    opendatahub.io/notebook-image-name: "Minimal Python"
+    opendatahub.io/notebook-image-desc: "Jupyter notebook image with minimal dependency set to start experimenting with Jupyter environment."
+    opendatahub.io/notebook-image-order: "10"
+  name: s2i-minimal-notebook
```
Comment (samuelvl) [inline jupyter-notebooks/base/minimal-notebook-imagestream.yaml:11]:
```suggestion
  name: jupyter-minimal-notebook
```

File: jupyter-notebooks/base/minimal-notebook-imagestream.yaml [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+    local: true
+  tags:
+  - annotations:
+      opendatahub.io/notebook-software: '[{"name":"Python","version":"v3.8"}]'
+      opendatahub.io/notebook-python-dependencies: '[{"name":"JupyterLab","version": "3.2"}, {"name": "Notebook","version": "6.4"}]'
+      openshift.io/imported-from: quay.io/opendatahub/notebooks
+      opendatahub.io/default-image: "true"
+    from:
+      kind: DockerImage
+      # in a downstream build "latest" will be replaced
```
Comment (samuelvl) [inline jupyter-notebooks/base/minimal-notebook-imagestream.yaml:23]:
```suggestion
```

File: jupyter-notebooks/base/minimal-notebook-imagestream.yaml [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+      name: quay.io/opendatahub/notebooks@sha256:db08b64d259931c2849a8c1ac1550eb2d52cbc2599b56dbe0477cbb707857a96
+    # in a downstream build "rhodsversion" will be replaced
```
Comment (samuelvl) [inline jupyter-notebooks/base/minimal-notebook-imagestream.yaml:25]:
```suggestion
```

File: jupyter-notebooks/base/pytorch-notebook-imagestream.yaml [hunk 10]
```diff
@@ -0,0 +1,27 @@
+apiVersion: image.openshift.io/v1
```
Comment (samuelvl) [inline jupyter-notebooks/base/pytorch-notebook-imagestream.yaml:1]:
```suggestion
---
apiVersion: image.openshift.io/v1
```

File: jupyter-notebooks/base/pytorch-notebook-imagestream.yaml [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+kind: ImageStream
+metadata:
+  labels:
+    opendatahub.io/notebook-image: "true"
+  annotations:
+    opendatahub.io/notebook-image-url: "https://github.com/opendatahub-io/notebooks/blob/main/jupyter/pytorch/ubi8-python-3.8"
+    opendatahub.io/notebook-image-name: "PyTorch"
+    opendatahub.io/notebook-image-desc: "Jupyter notebook image with PyTorch libraries and dependencies to start experimenting with advanced AI/ML notebooks."
+    opendatahub.io/notebook-image-order: "40"
```
Comment (samuelvl) [inline jupyter-notebooks/base/pytorch-notebook-imagestream.yaml:10]:
```suggestion
    opendatahub.io/notebook-image-order: "4"
```

File: jupyter-notebooks/base/pytorch-notebook-imagestream.yaml [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+  lookupPolicy:
+    local: true
+  tags:
+  - annotations:
+        opendatahub.io/notebook-software: '[{"name":"Python","version":"v3.8"},{"name":"PyTorch","version":"1.8"}]'
+        opendatahub.io/notebook-python-dependencies: '[{"name":"PyTorch","version":"1.8"},{"name":"Tensorboard","version":"2.6"},{"name":"Boto3","version":"1.17"},{"name":"Kafka-Python","version":"2.0"},{"name":"Matplotlib","version":"3.4"},{"name":"Numpy","version":"1.19"},{"name":"Pandas","version":"1.2"},{"name":"Scikit-learn","version":"0.24"},{"name":"Scipy","version":"1.6"}]'
+        openshift.io/imported-from: quay.io/opendatahub/notebooks
+    from:
+      kind: DockerImage
+      # in a downstream build "latest" will be replaced
```
Comment (samuelvl) [inline jupyter-notebooks/base/pytorch-notebook-imagestream.yaml:22]:
```suggestion
```

File: jupyter-notebooks/base/pytorch-notebook-imagestream.yaml [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+      name: quay.io/opendatahub/notebooks@sha256:050362cf80da80616ab7911566a9c32ebeafcd67ddd80c100f7edbeabc411529
+    # in a downstream build "rhodsversion" will be replaced
```
Comment (samuelvl) [inline jupyter-notebooks/base/pytorch-notebook-imagestream.yaml:24]:
```suggestion
```

File: jupyter-notebooks/base/pytorch-notebook-imagestream.yaml [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+    name: "py3.8-cuda-11.4.2-2"
```
Comment (samuelvl) [inline jupyter-notebooks/base/pytorch-notebook-imagestream.yaml:25]:
```suggestion
    name: "py3.8-v1"
```

File: jupyter-notebooks/base/pytorch-notebook-imagestream.yaml [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+kind: ImageStream
+metadata:
+  labels:
+    opendatahub.io/notebook-image: "true"
+  annotations:
+    opendatahub.io/notebook-image-url: "https://github.com/opendatahub-io/notebooks/blob/main/jupyter/pytorch/ubi8-python-3.8"
+    opendatahub.io/notebook-image-name: "PyTorch"
+    opendatahub.io/notebook-image-desc: "Jupyter notebook image with PyTorch libraries and dependencies to start experimenting with advanced AI/ML notebooks."
+    opendatahub.io/notebook-image-order: "40"
+  name: pytorch
```
Comment (samuelvl) [inline jupyter-notebooks/base/pytorch-notebook-imagestream.yaml:11]:
```suggestion
  name: jupyter-pytorch-gpu-notebook
```

File: jupyter-notebooks/base/tensorflow-notebook-imagestream.yaml [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+kind: ImageStream
+metadata:
+  labels:
+    opendatahub.io/notebook-image: "true"
+  annotations:
+    opendatahub.io/notebook-image-url: "https://github.com/opendatahub-io/notebooks/blob/main/jupyter/tensorflow/ubi8-python-3.8"
+    opendatahub.io/notebook-image-name: "TensorFlow"
+    opendatahub.io/notebook-image-desc: "Jupyter notebook image with TensorFlow libraries and dependencies to start experimenting with advanced AI/ML notebooks."
+    opendatahub.io/notebook-image-order: "50"
+  name: tensorflow
```
Comment (samuelvl) [inline jupyter-notebooks/base/tensorflow-notebook-imagestream.yaml:11]:
```suggestion
  name: jupyter-tensorflow-gpu-notebook
```

File: jupyter-notebooks/base/tensorflow-notebook-imagestream.yaml [hunk 17]
```diff
@@ -0,0 +1,27 @@
...
+  lookupPolicy:
+    local: true
+  tags:
+  - annotations:
+        opendatahub.io/notebook-software: '[{"name":"Python","version":"v3.8"},{"name":"TensorFlow","version":"2.7"}]'
+        opendatahub.io/notebook-python-dependencies: '[{"name":"TensorFlow","version":"2.7"},{"name":"Tensorboard","version":"2.6"},{"name":"Boto3","version":"1.17"},{"name":"Kafka-Python","version":"2.0"},{"name":"Matplotlib","version":"3.4"},{"name":"Numpy","version":"1.19"},{"name":"Pandas","version":"1.2"},{"name":"Scikit-learn","version":"0.24"},{"name":"Scipy","version":"1.6"}]'
+        openshift.io/imported-from: quay.io/opendatahub/notebooks
+    from:
+      kind: DockerImage
+      # in a downstream build "latest" will be replaced
```
Comment (samuelvl) [inline jupyter-notebooks/base/tensorflow-notebook-imagestream.yaml:22]:
```suggestion
```

File: jupyter-notebooks/base/tensorflow-notebook-imagestream.yaml [hunk 18]
```diff
@@ -0,0 +1,27 @@
...
+      name: quay.io/opendatahub/notebooks@sha256:2eebb34ec58d08d52d1dca6dcf07b52a0f1dd7a3468b7ca1299d803a0c6eebea
+    # in a downstream build "rhodsversion" will be replaced
```
Comment (samuelvl) [inline jupyter-notebooks/base/tensorflow-notebook-imagestream.yaml:24]:
```suggestion
```

File: jupyter-notebooks/base/tensorflow-notebook-imagestream.yaml [hunk 19]
```diff
@@ -0,0 +1,27 @@
...
+    name: "py3.8-cuda-11.4.2-2"
```
Comment (samuelvl) [inline jupyter-notebooks/base/tensorflow-notebook-imagestream.yaml:25]:
```suggestion
    name: "py3.8-v1"
```

File: tests/resources/notebook-controller/notebooks/generic-data-science-oauth-notebook.yaml [hunk 20]
```diff
@@ -1,16 +1,15 @@
----
```
Comment (samuelvl) [inline tests/resources/notebook-controller/notebooks/generic-data-science-oauth-notebook.yaml:1]: Do not remove yaml header

File: tests/resources/notebook-controller/notebooks/generic-data-science-oauth-notebook.yaml [hunk 21]
```diff
@@ -1,16 +1,15 @@
...
   annotations:
     notebooks.opendatahub.io/inject-oauth: "true"
 spec:
   template:
     spec:
       containers:
-        - name: thoth-minimal-oauth-notebook
-          image: image-registry.openshift-image-registry.svc:5000/opendatahub/s2i-minimal-notebook:v0.3.0-py38
+        - name: jupyter-datascience-ubi8-python-3-8
+          image: qquay.io/opendatahub/notebooks:jupyter-datascience-ubi8-python-3.8
```
Comment (samuelvl) [inline tests/resources/notebook-controller/notebooks/generic-data-science-oauth-notebook.yaml:12]: We want to test the images imported by the previous image streams, not those in quay.io. For example:

```suggestion
          image: image-registry.openshift-image-registry.svc:5000/opendatahub/jupyter-datascience-notebook:py3.8-v1
```

Review the other notebooks too.

**Review by samuelvl — CHANGES_REQUESTED** [review]

**Comment by samuelvl** [comment]
@atheo89 Tests are failing because of:

```
 Standard error from the command:
./run_robot_test.sh: line 194: lsb_release: command not found
You are using pip version 21.3.1, however version 22.3.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
[ERROR] PID 454: test/lib/log/stacktrace.sh:0: `return "${return_code}"` exited with status 1.
[INFO] 		Stack Trace: 
[INFO] 		  1: test/lib/log/stacktrace.sh:0: `return "${return_code}"`
[INFO] 		  2: test/lib/util/trap.sh:65: os::log::stacktrace::print
[INFO] 		  3: test/lib/cmd.sh:10: os::util::trap::err_handler
[INFO] 		  4: operator-tests/odh-manifests/basictests/dashboard.sh:66: os::cmd::expect_success
[INFO] 		  5: operator-tests/odh-manifests/basictests/dashboard.sh:74: test_odh_dashboard_ui
[INFO]   Exiting with code 1.
failed: /root/peak/operator-tests/odh-manifests/basictests/dashboard.sh 
```

This is not related with your PR, could you take a look? @LaVLaS

**Comment by LaVLaS** [comment]
@atheo89 I will be merging this today after I finish and manual test while I finish a PR to resolve the CI test issues

**Comment by atheo89** [comment]
>Can you change the component folder to `notebook-images` or was the `notebooks-images` intentional?
@LaVLaS I push the changes (it was a typo error)

**Review by LaVLaS — APPROVED** [review]
/lgtm

**Pull request merged**